docs/deploy/javascript.rst

```diff
@@ -32,52 +32,163 @@ Note that the `WebLLM prebuilt webpage <https://webllm.mlc.ai/#chat-demo>`__ abo
-      setLabel("init-label", report.text);
-    });
+  If you added your conversation template in ``src/conversation.ts``, you need to build from source
+  following the instruction in `the WebLLM repo's README <https://github.com/mlc-ai/web-llm?tab=readme-ov-file#build-webllm-package-from-source>`_. 
 
-    await chat.reload("Llama-2-7b-chat-hf-q4f32_1");
+  Alternatively, you could use the ``"custom"`` conversation template so that you can pass in
+  your own ``ConvTemplateConfig`` in runtime without having to build the package from source.
 
-    const generateProgressCallback = (_step: number, message: string) => {
-      setLabel("generate-label", message);
-    };
+**Step 3 Adding as ModelRecord**
 
-    const prompt0 = "What is the capital of Canada?";
-    setLabel("prompt-label", prompt0);
-    const reply0 = await chat.generate(prompt0, generateProgressCallback);
-    console.log(reply0);
+Finally, we modify the code snippet for
+`get-started <https://github.com/mlc-ai/web-llm/blob/main/examples/get-started/src/get_started.ts>`__
+pasted above.
 
-    const prompt1 = "Can you write a poem about it?";
-    setLabel("prompt-label", prompt1);
-    const reply1 = await chat.generate(prompt1, generateProgressCallback);
-    console.log(reply1);
+We simply specify the Huggingface link as ``model_url``, while reusing the ``model_lib_url`` for 
+``Mistral-7B``. Note that we need the suffix to be ``/resolve/main/``.
 
-    console.log(await chat.runtimeStatsText());
+.. code:: typescript
+
+  const myAppConfig: AppConfig = {
+    model_list: [
+      // Other records here omitted...
+      {
+        "model_url": "https://huggingface.co/mlc-ai/WizardMath-7B-V1.1-q4f16_1-MLC/resolve/main/",
+        "local_id": "WizardMath-7B-V1.1-q4f16_1",
+        "model_lib_url": "https://raw.githubusercontent.com/mlc-ai/binary-mlc-llm-libs/main/Mistral-7B-Instruct-v0.2-q4f16_1-sw4k_cs1k-MLC-webgpu.wasm",
+        "required_features": ["shader-f16"],
+      },
+    ]
   }
 
-  main();
+  const selectedModel = "WizardMath-7B-V1.1-q4f16_1"
+  await chat.reload(selectedModel, undefined, myAppConfig);
+
+Now, running the ``get-started`` example will use the ``WizardMath`` model you just added.
+
+
+Bring Your Own Model Library
+----------------------------
+
+A model library is specified by:
+
+ - The model architecture (e.g. ``llama-2``, ``gpt-neox``)
+ - Quantization (e.g. ``q4f16_1``, ``q0f32``)
+ - Metadata (e.g. ``context_window_size``, ``sliding_window_size``, ``prefill-chunk-size``), which affects memory planning
+ - Platform (e.g. ``cuda``, ``webgpu``, ``iOS``)
+
+In cases where the model you want to run is not compatible with the provided MLC
+prebuilt model libraries (e.g. having a different quantization, a different
+metadata spec, or even a different model architecture), you need to build your
+own model library.
+
+Follow the page :ref:`compile-model-libraries`, specifically the ``WebGPU`` option. The page
```

Comment:
Would be great to copy the code over to here, so readers to not need to jump back and forth. Then add a reference to say if they would like to know more, they should look at the other page